ui/src/contexts/Application.jsx

```diff
@@ -97,7 +97,17 @@ function watchVault(id, dispatch, offerStatus) {
   }
 
   async function watch() {
-    const { vault, asset } = await E(walletP).getPublicNotifiers(id);
+    let vault;
+    let asset;
+    try {
+      const notifiers = await E(walletP).getPublicNotifiers(id);
+      vault = notifiers.vault;
+      asset = notifiers.asset;
```

Comment:
I think this would work and be more clear about what's going on:
```suggestion
      ({ asset, vault }) = notifiers.vault);
```

another option is to make a helper function that could be used here,
```
const { vault, asset, errored } = await getNotifiersOrError(id);
if (errored) {
  console.error('Could not get notifiers', id, err);
  dispatch(updateVault({ id, vault: { status: VaultStatus.ERROR, err } }));
  return;
}
```

Reply:
Thanks, went with the first option. I was not aware of that destructuring syntax for `let` vars.